### PR TITLE
add workloadd events to pytorch

### PR DESCRIPTION
**Pull request by fenypatel99**
Summary: add workloadd events to pytorch

Test Plan: CIs

Differential Revision: D57914472

### CODE REVIEW

File: torch/csrc/profiler/kineto_shim.cpp [hunk 1]
```diff
@@ -358,6 +357,20 @@ c10::DeviceType deviceTypeFromActivity(libkineto::ActivityType activity_type) {
       }();
       return device_type;
     }
+    // TODO: T151322015
+    case libkineto::ActivityType::MTIA_CCP_EVENTS:
+    case libkineto::ActivityType::MTIA_WORKLOADD: {
+      // PrivateUse1 kineto backend reuse above ActivityTypes,
+      // If PrivateUse1 backend enabled, this should return
+      // c10::DeviceType::PrivateUse1.
+      c10::DeviceType device_type = []() {
```
Comment (briancoutinho) [inline torch/csrc/profiler/kineto_shim.cpp:366]: why is this in a closure, lambda? did you mean to make this static

File: torch/csrc/profiler/kineto_shim.cpp [hunk 2]
```diff
@@ -344,9 +345,7 @@ c10::DeviceType deviceTypeFromActivity(libkineto::ActivityType activity_type) {
     case libkineto::ActivityType::CONCURRENT_KERNEL:
     case libkineto::ActivityType::CUDA_SYNC:
     case libkineto::ActivityType::GPU_USER_ANNOTATION:
-    case libkineto::ActivityType::CUDA_PROFILER_RANGE:
-    // TODO: T151322015
-    case libkineto::ActivityType::MTIA_CCP_EVENTS: {
+    case libkineto::ActivityType::CUDA_PROFILER_RANGE: {
```
Comment (briancoutinho) [inline torch/csrc/profiler/kineto_shim.cpp:348]: is this intentional? cuda profiler range is range profiler

Comment (fenypatel99) [reply torch/csrc/profiler/kineto_shim.cpp:348]: Hmm, I did not change the name. It was CUDA_PROFILER_RANGE

File: torch/csrc/profiler/kineto_shim.cpp (hunk 1 above)
Comment (fenypatel99) [reply torch/csrc/profiler/kineto_shim.cpp:366]: To keep it synchronous with CUDA, I used that, no specific intention here

**Review by sraikund16 — APPROVED** [review]

**Pull request closed without merging**